source/code/plugins/npmd_config_lib.rb

```diff
@@ -632,16 +632,18 @@ def self.getEpmHashFromJson(text)
                     _epmRules = {"Rules" => []}
                     # Check all tests related to current agent id and push their configurations to current agent
                     _testIds = _h[EpmAgentInfoTag][_agentId]
+                    return if _testIds.nil?
+
                     _testIds.each do |testId|
                         _test = _h[EpmTestInfoTag][testId]
                         _rule = Hash.new
                         _rule["ID"] = testId
                         _rule["Name"] = _test["Name"]
                         _rule["Poll"] = _test["Poll"]
                         _rule["AppThresholdLoss"] = _test["AppThreshold"].has_key?("Loss") ? _test["AppThreshold"]["Loss"] : "-2"
-                        _rule["AppThresholdLatency"] = _test["AppThreshold"]["Latency"]
-                        _rule["NetworkThresholdLoss"] = _test["NetworkThreshold"]["Loss"]
-                        _rule["NetworkThresholdLatency"] = _test["NetworkThreshold"]["Latency"]
+                        _rule["AppThresholdLatency"] = _test["AppThreshold"].has_key?("Latency") ? _test["AppThreshold"]["Latency"] : "-2.0"
```

Comment:
you should also check here whethere test["AppThreshold"] is nil or not without making any assumptions that they will always be there

Reply:
Validation added.